sw/device/silicon_creator/lib/drivers/otbn.c

```diff
@@ -57,8 +58,67 @@ static rom_error_t check_offset_len(uint32_t offset_bytes, size_t num_words,
   return kErrorOk;
 }
 
-void otbn_execute(void) {
-  abs_mmio_write32(kBase + OTBN_CMD_REG_OFFSET, kOtbnCmdExecute);
+/**
+ * Helper function for writing to OTBN's DMEM or IMEM.
+ *
+ * @param dest_addr Destination address.
+ * @param src Source buffer.
+ * @param num_words Number of words to copy.
+ */
+static void otbn_write(uint32_t dest_addr, const uint32_t *src,
+                       size_t num_words) {
+  // Start from a random index less than `num_words`.
+  size_t i = ((uint64_t)rnd_uint32() * (uint64_t)num_words) >> 32;
+  enum { kStep = 1 };
+  size_t iter_cnt = 0;
+  for (; launder32(iter_cnt) < num_words; ++iter_cnt) {
+    abs_mmio_write32(dest_addr + i * sizeof(uint32_t), src[i]);
+    i += kStep;
+    if (i >= num_words) {
+      i -= num_words;
+    }
```

Comment:
Are we at all worried that someone could glitch this check and write to some out-of-range address? I suppose that it wouldn't be very helpful to them, especially if they couldn't control which limb exactly they were glitching.

Reply:
I'm tempted to say not really since the initial value of `i` is random and it would be pretty hard to do this in a controlled way.

@moidx any thoughts?

Reply:
We are concerned about writing to unintended random addresses, but we assume that the attack is hard to reproduce given the initial random value.  

One thing we could consider is to add `launder(i) < num_words` to the for condition.

Reply:
Done, (similar to `sigverify_rsa_key_get()`).